### PR TITLE
feat(Core): Update core dependencies and logic

### PR DESCRIPTION
**Pull request by Coloud1**

### CODE REVIEW

File: lib/app/util/enum/swagger_path_request_type.dart [hunk 1]
```diff
@@ -3,6 +3,7 @@ enum SwaggerPathRequestType {
   post,
   put,
   delete,
+  patch,
```
Comment (Coloud1) [inline lib/app/util/enum/swagger_path_request_type.dart:6]: @cozvtieg9, one swagger did not work properly and fell on an unimplemented orElse

File: lib/data/model/swagger/types/swagger_type.dart [hunk 2]
```diff
@@ -24,6 +24,12 @@ sealed class SwaggerType {
   String? getFileImportName(DataFileType fileType);
 
   String? getDefaultReturnType(DataFileType fileType);
+
+  String? getFullFileImport(String projectName, DataFileType fileType) {
+    final importName = getFileImportName(fileType);
```
Comment (Coloud1) [inline lib/data/model/swagger/types/swagger_type.dart:29]: @cozvtieg9, додав перевірку на null, щоб не ламати твою існуючу логіку перевірки на import == null || imports.contains(import);

File: lib/domain/entity/component/data_object_component.dart [hunk 3]
```diff
@@ -150,6 +150,9 @@ class DataObjectComponent with _$DataObjectComponent {
     }
     codeLines.addNewLine();
     final classNamePrefix = fileReference.getTypeDeclaration(DataFileType.none);
+    final objects =
+        _getMapperObjectVariablesContent(MapperType.mapEntityToRequest);
+    final classModifier = objects.isEmpty ? 'const ' : '';
```
Comment (Coloud1) [inline lib/domain/entity/component/data_object_component.dart:155]: @cozvtieg9, added logic to the const constructor to have fewer hints in the analyzer

File: lib/domain/entity/component/source_component.dart [hunk 4]
```diff
@@ -213,46 +208,47 @@ class SourceComponent with _$SourceComponent {
...
   String _buildSourceImports(String projectName) {
     final imports = List<String>.empty(growable: true);
-    for (var request in requests) {
+    for (final request in requests) {
       ///build response imports
       ///Add response body import
-      final responseFileImport =
-          request.response.type.getFileImportName(DataFileType.response);
+      final responseFileImport = request.response.type
+          .getFullFileImport(projectName, DataFileType.response);
```
Comment (Coloud1) [inline lib/domain/entity/component/source_component.dart:233]: @cozvtieg9,  замінив getFileImportName на getFullFileImport, щоб підтримувати пекеджі. Назву можна замінити, якщо не дуже підходить

File: lib/domain/service/base/class_builder/freezed_class_builder.dart [hunk 5]
```diff
@@ -31,31 +32,35 @@ class FreezedClassBuilder extends ClassBuilder {
...
+      ..add("import 'package:freezed_annotation/freezed_annotation.dart';")
+      ..addAll(_imports)
+      ..addNewLine()
+      ..add("part '$classPartImport.freezed.dart';")
+      ..addNewLine()
+      ..add('@freezed')
+      ..add('class $classFullName with _\$$classFullName {');
     if (_baseConstructorProperties.isEmpty) {
-      lines.add('factory $classFullName() = _$classFullName;');
+      lines.add('const factory $classFullName() = _$classFullName;');
```
Comment (Coloud1) [inline lib/domain/service/base/class_builder/freezed_class_builder.dart:44]: @cozvtieg9, I mentioned above that I added const constructors, this const factory has been updated for this

File: lib/domain/service/component_generator/component_generator_service.dart [hunk 6]
```diff
@@ -86,15 +86,15 @@ class ComponentGeneratorService
     final declarationFilePath =
         sourceComponent.getDeclarationFilePath(projectLibFolder);
 
-    final declarationBody = sourceComponent.getDeclarationBody(projectName);
+    final declarationBody = sourceComponent.getSourceDeclarationBody(projectName);
```
Comment (Coloud1) [inline lib/domain/service/component_generator/component_generator_service.dart:89]: @cozvtieg9, changed the name from getDecralationBody to getSourceDeclarationBody to make it easier to understand where the surfaces are and where the repos are

File: lib/presentation/screen/generation_screen/bloc/generation_screen_bloc.dart [hunk 7]
```diff
@@ -154,7 +154,7 @@ class GenerationScreenBloc extends BaseBloc<GenerationScreenEvent,
 
       if (!state.config.graphql) {
         await Directory(
-          '${state.config.projectPath}/${state.config.projectName}/lib/core/arch/data/remote/clients/graph_ql',
+          '${state.config.projectPath}/${state.config.projectName}/lib/core/arch/data/remote/graph_ql',
```
Comment (Coloud1) [inline lib/presentation/screen/generation_screen/bloc/generation_screen_bloc.dart:157]: @cozvtieg9, removed the clients directory

File: bricks/flutter_clean_base/__brick__/{{project_name.snakeCase()}}/lib/data/source/remote/firebase/auth/firebase_auth_source_impl.dart [hunk 8]
```diff
@@ -70,7 +68,7 @@ class FirebaseAuthSourceImpl extends FirebaseAuthSource {
         return const DataResponse.unauthorized();
       }
     } on FirebaseAuthException catch (e) {
-      return DataResponse.firebaseError(e.code);
+      return DataResponse.undefinedError(e);
```
Comment (Coloud1) [inline bricks/flutter_clean_base/__brick__/{{project_name.snakeCase()}}/lib/data/source/remote/firebase/auth/firebase_auth_source_impl.dart:71]: @cozvtieg9,  I removed the firebaseError. There is no way to extend the freezed class

**Review by cozvtieg9 — APPROVED** [review]

**Pull request merged**